doc/lua_api.txt

```diff
@@ -3916,6 +3916,13 @@ The Biome API is still in an experimental phase and subject to change.
     -- ^ Minimum and maximum `y` positions these decorations can be generated at.
     -- ^ This parameter refers to the `y` position of the decoration base, so
     --   the actual maximum height would be `height_max + size.Y`.
+        spawn_by = "default:water",
+    --  ^ Node that the decoration only spawns next to.
+    --  ^ The neighbours checked are the 16 nodes level with and surrounding
+    --  ^ these 2 nodes: the surface node and the node above the surface node.
```

Comment:
Indeed i was having problems trying to word this, your suggestion is better.

By the way the second node checked has Y = surface + 1.
I left the hole because 'spawn by' means 'spawn horizontally by': In the case of a simpe decoration the spawn point is the surface node, but the decoration is placed in the node above that (and upwards in the case of a cactus or papyris), so we don't check that node above.
In the case of a schematic decoration the schematic is placed at the spawn point (the surface node) and often extends upwards from there.